conn.go

```diff
@@ -0,0 +1,190 @@
+package p2pd
+
+import (
+	"context"
+	"errors"
+	"io"
+	"net"
+	"time"
+
+	pb "github.com/libp2p/go-libp2p-daemon/pb"
+
+	ggio "github.com/gogo/protobuf/io"
+	inet "github.com/libp2p/go-libp2p-net"
+	peer "github.com/libp2p/go-libp2p-peer"
+	pstore "github.com/libp2p/go-libp2p-peerstore"
+	proto "github.com/libp2p/go-libp2p-protocol"
+	ma "github.com/multiformats/go-multiaddr"
+)
+
+const DefaultTimeout = 60 * time.Second
+
+func (d *Daemon) handleConn(c net.Conn) {
+	defer c.Close()
+
+	r := ggio.NewDelimitedReader(c, inet.MessageSizeMax)
+	w := ggio.NewDelimitedWriter(c)
+
+	for {
+		var req pb.Request
+
+		err := r.ReadMsg(&req)
+		if err != nil {
+			if err != io.EOF {
+				log.Debugf("Error reading message: %s", err.Error())
+			}
+			return
+		}
+
+		switch *req.Type {
```

Comment:
idk if it's necessary for this PR, but we can definitely abstract this to support handler functions of the variety

```golang
func (req *pb.Request) *pb.Response
```

Reply:
ah well it seems it doesn't perfectly map

Reply:
yeah, the stream breaks the abstraction. Also, notifications later on.